### DISCUSSION TITLE
Failed to invoke the specified Python3 path

### DISCUSSION THREAD

**Discussion by forever-happy-h** [sylabs/singularity#2250] [Q&A]
Hello,

Whenever I need to invoke my desired Python path, the following situation occurs.

$ singularity exec AF_environment.sif /data/mini3/envs/alphafold/bin/python /data/protien/FoldDock/src/sequence_extract.py
INFO: Converting SIF file to temporary sandbox...
FATAL: stat /data2/lvhaohao/mini3/envs/alphafold/bin/python: no such file or directory

And when I don't specify the Python path, Singularity defaults to invoking python3 from /opt/conda/bin/python3, but that file doesn't exist for me.
Afterwards, the following error message will appear.

$ singularity exec AF_environment.sif python3 "/data/protien/FoldDock/src/sequence_extract.py"
INFO: Converting SIF file to temporary sandbox...
/opt/conda/bin/python3: can't open file '/data/protien/FoldDock/src/sequence_extract.py': [Er] No such file or directory

Thank you in advance.

**Comment by dtrudg** [comment]
Hello. This isn't a bug in Singularity, but is a usage question. Therefore it's better as a GitHub discussion, rather than an issue. I'll transfer it to a discussion now.

**Comment by dtrudg** [comment]
Hello - Singularity doesn't choose which python to invoke here. It will depend on the content of the `sequence_extract.py` and the general configuration of the container you are using.

You will need to give some more information about the container and the script in order for others to assist.
